vcr/pe/matcher_test.go

```diff
@@ -0,0 +1,391 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package pe
+
+import (
+	"encoding/json"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+	"testing"
+)
+
+const testPresentationDefinition = `
+{
+  "id": "Definition requesting NutsOrganizationCredential",
+  "input_descriptors": [
+	{
+	  "id": "some random ID",
+	  "name": "Organization matcher",
+	  "purpose": "Finding any organization in CareTown starting with 'Care'",
+	  "constraints": {
+		"fields": [
+		  {
+			"path": [
+			  "$.credentialSubject.organization.city"
+			],
+			"filter": {
+			  "type": "string",
+			  "const": "IJbergen"
+			}
+		  },
+		  {
+			"path": [
+			  "$.credentialSubject.organization.name"
+			],
+			"filter": {
+			  "type": "string",
+			  "pattern": "care"
+			}
+		  },
+		  {
+			"path": [
+			  "$.type"
+			],
+			"filter": {
+			  "type": "string",
+			  "const": "NutsOrganizationCredential"
+			}
+		  }
+		]
+	  }
+	}
+  ],
+  "format": {
+    "jwt_vc": {
+      "alg": ["ES256K", "ES384"]
+    },
+	"ldp_vc": {
+      "proof_type": [
+	    "JsonWebSignature2020"
+	  ]
+	}
+  } 
+}
+`
+
+var testCredentialString = `
+{
+  "type": "VerifiableCredential",
+  "credentialSubject": {
+	"field": "value"
+  }
+}`
+
+func TestMatch(t *testing.T) {
```

Comment:
You have many tests asserting small individual part, but just 1 for the function (`Match`) that'll actually be called by the rest of the code (which is to come). For logical test coverage and understand of the reader, it'd help to have more tests for `Match`. Since tests for `Match` don't change when you change the internals of how it works (all the other smaller non-exported functions), but the other small tests will.

Reply:
The smaller functions won't change unless the spec changes. They are a 1-on-1 copy of the bullet points of the spec. A `presentation_definition` is quite verbose so you probably want individual tests to use the struct directly which basically is the same as the smaller tests right not. I would rather add another larger real world test case when we have one.